src/demo-app/a11y/list/list-a11y.html

```diff
@@ -0,0 +1,71 @@
+<div class="demo-list">
+  <section>
+    <h2> Seasoning </h2>
+    <p>Showing a non-interactive list of seasonings.</p>
+    <md-list>
+      <md-list-item *ngFor="let item of items"> {{item}} </md-list-item>
+    </md-list>
+  </section>
+
+
+  <section>
+    <h2>Mailbox navigation</h2>
+    <p>Showing a navigation list with links to google search, and "more information" icon button</p>
+    <md-nav-list>
```

Comment:
`md-nav-list` is only meant to contain `<a md-list-item>` children.

This makes me realize that we probably need tweaks as part of this:
* Removing the existing roles and saying that, by default, `md-list` is assumed to be presentational, and add instructions to add the roles for non-interactive content (same as grid-list).
* Give `md-nav-list` `role="navigation"`

Thoughts?